src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/ProviderBase/DbConnectionFactory.cs

```diff
@@ -65,8 +65,12 @@ internal int ObjectID
 
         public void ClearAllPools()
         {
-            IntPtr hscp;
-            Bid.ScopeEnter(out hscp, "<prov.DbConnectionFactory.ClearAllPools|API> ");
+            long scopeID = 0;
+            if (SqlClientEventSource.Log.IsScopeEnabled())
```

Comment:
Can we move this check inside the logging function, i.e. `ScopeEnter(...)` The usage pattern is that if enabled then log/trace etc. Instead if we just let the ScopeEnter/Trace etc to do this check, there is much lesser scope for programming error like, developer logging without checking if the logging is enabled.

That's really the principle behind other logging frameworks as well, logging can be invoked, but the underlying framework checks if the logs should be actually emitted or not.

Reply:
Basically follow the same checking patter that `SqlClientEventSource.Log.ScopeLeave(scopeID);` is following. It checks internally if the event should be emitted or not.